src/ray/raylet/node_manager.cc

```diff
@@ -814,6 +816,25 @@ void NodeManager::HandleGetObjectsInfo(const rpc::GetObjectsInfoRequest &request
       /*on_all_replied*/ [total, reply]() { reply->set_total(*total); });
 }
 
+void NodeManager::HandleGetTaskResult(const rpc::GetTaskResultRequest &request,
+                          rpc::GetTaskResultReply *reply,
+                          rpc::SendReplyCallback send_reply_callback) {
+  const TaskID task_id = TaskID::FromBinary(request.task_id());
+  RAY_LOG(DEBUG) << "Received a HandleGetTaskResult request for task " << task_id;
+  
+  auto it = task_failure_reasons_.find(task_id);
```

Comment:
Does it also cover cases like worker failure errors?

Reply:
confirmed we will do this in follow up